packages/typescript/src/index.ts

```diff
@@ -142,11 +142,15 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
 
       const tsBuildInfoPath = ts.getTsBuildInfoEmitOutputFilePath(parsedOptions.options);
       if (tsBuildInfoPath) {
-        this.emitFile({
-          type: 'asset',
-          fileName: normalizePath(path.relative(outputOptions.dir!, tsBuildInfoPath)),
-          source: emittedFiles.get(tsBuildInfoPath)
-        });
+        const tsBuildInfoSource = emittedFiles.get(tsBuildInfoPath);
+        // https://github.com/rollup/plugins/issues/681
+        if (tsBuildInfoSource) {
```

Comment:
> The plugin now checks if TypeScript has emitted a tsbuildinfo file. If it hasn't, the plugin doesn't emit the file, which prevents the TypeError from occurring.

This `if` block here seems to be opposite of what you're describing in the PR description. Which one is the correct behavior?

Reply:
If and only if TypeScript has emitted a tsbuildinfo file (`if (tsBuildInfoSource)`), the plugin emits the tsbuildinfo file.	This is the same behaviour described in the PR description, but I'll update it to be a bit clearer. Sorry for the confusion.

Reply:
Thanks for the further explanation, and no worries.